libnmstate/nm/bridge.py

```diff
@@ -165,36 +176,57 @@ def _get_bridge_setting(nmdev):
     return bridge_setting
 
 
-def _get_bridge_ports_info(port_profiles, vlan_filtering_enabled=False):
-    ports_info = []
-    for p in port_profiles:
-        port_info = _get_bridge_port_info(p)
+def _get_bridge_ports_info(
+    port_profiles_by_name, port_runtime_names, vlan_filtering_enabled=False
+):
+    ports_info_by_name = {
+        name: _get_bridge_port_name_info(name) for name in port_runtime_names
+    }
+
+    for name, p in port_profiles_by_name.items():
+        port_info = ports_info_by_name.get(name, {})
         if port_info:
             if vlan_filtering_enabled:
                 bridge_vlan_config = p.get_setting_bridge_port().props.vlans
                 port_vlan = PortVlanFilter()
                 port_vlan.import_from_bridge_settings(bridge_vlan_config)
                 port_info[LB.Port.VLAN_SUBTREE] = port_vlan.to_dict()
-            ports_info.append(port_info)
-    return ports_info
-
+    return list(ports_info_by_name.values())
 
-def _get_bridge_port_info(port_profile):
-    """Report port information."""
 
-    port_setting = port_profile.get_setting_bridge_port()
-    return {
-        LB.Port.NAME: port_profile.get_interface_name(),
-        LB.Port.STP_PRIORITY: port_setting.props.priority,
-        LB.Port.STP_HAIRPIN_MODE: port_setting.props.hairpin_mode,
-        LB.Port.STP_PATH_COST: port_setting.props.path_cost,
-    }
-
-
-def _get_slave_profiles(master_device):
-    slave_profiles = []
+def _get_slave_names_profiles(master_device):
+    slaves_profiles_by_name = {}
     for dev in master_device.get_slaves():
         active_con = connection.get_device_active_connection(dev)
         if active_con:
-            slave_profiles.append(active_con.props.connection)
-    return slave_profiles
+            slaves_profiles_by_name[
+                dev.get_iface()
+            ] = active_con.props.connection
+    return slaves_profiles_by_name
+
+
+def _get_bridge_port_name_info(port_runtime):
+    """Report port runtime information from sysfs."""
+    port = {LB.Port.NAME: port_runtime}
+    for option, option_sysfs in BRIDGE_PORT_NMSTATE_TO_SYSFS.items():
+        sysfs_path = f"/sys/class/net/{port_runtime}/brport/{option_sysfs}"
+        with open(sysfs_path) as f:
+            option_value = int(f.read())
+            if option == LB.Port.STP_HAIRPIN_MODE:
+                option_value = bool(option_value)
+        port[option] = option_value
+    return port
+
+
+def _get_slaves_runtime(master):
```

Comment:
How about `_get_slaves_names_from_runtime`?
`runtime` can be changed to `sysfs`.

Reply:
Done! Thanks!